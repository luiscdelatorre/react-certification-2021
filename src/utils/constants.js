const AUTH_STORAGE_KEY = 'wa_cert_authenticated';
const SEARCH_STORAGE_KEY = 'wa_cert_search';
const API_URL = `https://www.googleapis.com/youtube/v3`;

export { AUTH_STORAGE_KEY, SEARCH_STORAGE_KEY, API_URL };
