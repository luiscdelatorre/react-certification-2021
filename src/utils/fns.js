const random = (limit) => {
  return Math.floor(Math.random() * limit);
};

const formatDate = (dateString) => {
  const date = new Date(dateString);
  return date.toLocaleDateString();
};

export { random, formatDate };
