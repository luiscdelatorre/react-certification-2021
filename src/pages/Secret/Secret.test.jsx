import React from 'react';
import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import SecretPage from './Secret.page';
import SessionDataProvider from '../../providers/SessionData/SessionData.provider';
import { storage } from '../../utils/storage';
import { USER_STORAGE_KEY } from '../../utils/constants';
import UserMock from '../../mock/user.mock.json';

describe('Secret Page', () => {
  it('Check welcome message and back button in page', () => {
    storage.set(USER_STORAGE_KEY, UserMock);

    render(
      <BrowserRouter>
        <SessionDataProvider>
          <SecretPage />
        </SessionDataProvider>
      </BrowserRouter>
    );

    expect(screen.queryByText('Welcome Wizeline')).toBeInTheDocument();
    expect(screen.queryByText('Back to Home')).toBeInTheDocument();
  });

  it('Validates Secret snapshot', () => {
    const component = renderer.create(
      <BrowserRouter>
        <SessionDataProvider>
          <SecretPage />
        </SessionDataProvider>
      </BrowserRouter>
    );
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });
});
