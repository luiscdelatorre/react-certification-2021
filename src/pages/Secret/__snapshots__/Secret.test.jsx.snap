// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Secret Page Validates Secret snapshot 1`] = `
<section
  className="sc-kfzAmx bDjrZP"
>
  <h1
    className="sc-gsTCUz dkhwEY"
  >
    Welcome, voyager...
  </h1>
  <div
    className="sc-jrAGrp bnkjNS"
  >
    <iframe
      allow="accelerometer; autoplay; encrypted-media; gyroscope; picture-in-picture"
      allowFullScreen={true}
      className="sc-pFZIQ lkwIdY"
      frameBorder="0"
      src="https://www.youtube.com/embed/dQw4w9WgXcQ?controls=0&autoplay=1"
      title="rick roll"
    />
  </div>
  <a
    className="sc-fKFyDc dSWJJm"
    href="/"
    onClick={[Function]}
  >
    <svg
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 512 512"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M448 440a16 16 0 01-12.61-6.15c-22.86-29.27-44.07-51.86-73.32-67C335 352.88 301 345.59 256 344.23V424a16 16 0 01-27 11.57l-176-168a16 16 0 010-23.14l176-168A16 16 0 01256 88v80.36c74.14 3.41 129.38 30.91 164.35 81.87C449.32 292.44 464 350.9 464 424a16 16 0 01-16 16z"
      />
    </svg>
    Back to Home
  </a>
</section>
`;
